set.mm

```diff
@@ -652048,6 +652049,301 @@ of all sets ( ~ inex1g ).
+      cdif wi eldif cv cun imor biantrurd reim0b syl notbid eleq1 elrab 3bitr4d
+      bicomd recld 0red lenltd fveq2 breq2d orbi12d syl5bb elun ianor elrp rere
+      bicomi pm5.32i bitri xchbinxr rabbii unrab dfdif2 3eqtr4i eleq2d 3bitr2d
+      ) ABEFUCZGZBEGZBFGHZIZBUAJKLZBMJZKUBNZUDZWAVSWBOABEFUEPAWFBDUFZQGZHZDERZG
+      ZBKWGMJZSNZHZDERZGZTZBWJWOUGZGZVSWFWCHZWETAWQWCWEUHAWTWKWEWPABQGZHZVTXBIZ
+      WTWKAVTXBCUIAXBWTAXAWCAVTXAWCOCBUJUKULUPWKXCOAWIXBDBEWGBLZWHXAWGBQUMULUNP
+      UOAKWDSNZHZVTXFIZWEWPAVTXFCUIAWDKABCUQAURUSWPXGOAWNXFDBEXDWMXEXDWLWDKSWGB
+      MUTVAULUNPUOVBVCWSWQOABWJWOVDPAWRVRBWRVRLAWIWNTZDERWGFGZHZDERWRVRXHXJDEXH
+      WHWMIZXIXKHXHWHWMVEVHXIWHKWGSNZIXKWGVFWHXLWMWHWMXLWHWLWGKSWGVGVAUPVIVJVKV
+      LWIWNDEVMDEFVNVOPVPVQAVTWACUIUO $.
+  $}
+
+  $( Not before absnid: $)
+
+  $( Alternate expression for the absolute value of a real number.  Lemma for
+     ~ sqrtcval .  (Contributed by RP, 11-May-2024.) $)
+  reabsifneg $p |- ( A e. RR -> ( abs ` A ) = if ( A < 0 , -u A , A ) ) $=
+    ( cr wcel cc0 clt wbr cneg cif cabs cfv wa cle wceq wi ltle mpan2 imdistani
+    0re absnid eqcomd syl wn 0red id lenltd bicomd pm5.32i absid sylbi ifeqda )
+    ABCZADEFZAGZAHAIJZUKULUMAUNUKULKZUNUMUOUKADLFZKUNUMMUKULUPUKDBCULUPNRADOPQA
+    SUATUKULUBZKZUNAURUKDALFZKUNAMUKUQUSUKUSUQUKDAUKUCUKUDUEUFUGAUHUITUJT $.
+
+  $( Alternate expression for the absolute value of a real number.
+     (Contributed by RP, 11-May-2024.) $)
+  reabsifnpos $p |- ( A e. RR -> ( abs ` A ) = if ( A <_ 0 , -u A , A ) ) $=
+    ( cr wcel cc0 cle wbr cneg cif cabs cfv wa absnid eqcomd wn wceq wi 0re clt
+    ltnle ltle sylbird mpan imdistani absid syl ifeqda ) ABCZADEFZAGZAHAIJZUGUH
+    UIAUJUGUHKUJUIALMUGUHNZKZUJAULUGDAEFZKUJAOUGUKUMDBCZUGUKUMPQUNUGKUKDARFUMDA
+    SDATUAUBUCAUDUEMUFM $.
+
+  $( Alternate expression for the absolute value of a real number.
+     (Contributed by RP, 11-May-2024.) $)
+  reabsifpos $p |- ( A e. RR -> ( abs ` A ) = if ( 0 < A , A , -u A ) ) $=
+    ( cr wcel cc0 clt wbr cneg cif cabs cfv wa cle wceq 0re ltle mpan imdistani
+    wi absid eqcomd syl wn id 0red lenltd pm5.32i absnid sylbir ifeqda ) ABCZDA
+    EFZAAGZHAIJZUJUKAULUMUJUKKZUMAUNUJDALFZKUMAMUJUKUODBCUJUKUORNDAOPQASUATUJUK
+    UBZKZUMULUQUJADLFZKUMULMUJURUPUJADUJUCUJUDUEUFAUGUHTUIT $.
+
+  $( Alternate expression for the absolute value of a real number.
+     (Contributed by RP, 11-May-2024.) $)
+  reabsifnneg $p |- ( A e. RR -> ( abs ` A ) = if ( 0 <_ A , A , -u A ) ) $=
+    ( cr wcel cc0 cle wbr cneg cif cabs cfv wa absid eqcomd wn wi 0re clt ltnle
+    wceq ltle sylbird mpan2 imdistani absnid syl ifeqda ) ABCZDAEFZAAGZHAIJZUGU
+    HAUIUJUGUHKUJAALMUGUHNZKZUJUIULUGADEFZKUJUISUGUKUMUGDBCZUKUMOPUGUNKUKADQFUM
+    ADRADTUAUBUCAUDUEMUFM $.
+
+  $( Alternate expression for the absolute value of a real number.
+     (Contributed by RP, 22-May-2024.) $)
+  reabssgn $p |- ( A e. RR -> ( abs ` A ) = ( ( sgn ` A ) x. A ) ) $=
```

Comment:
Nice. Can you prove it for extended reals ?  Isn't there a shorter proof using reabsifnonneg ?

Reply:
df-abs and df-mul only want to work with complex numbers, so the infinities of the extended reals are out-of-bounds.

Reply:
Oh, I see. So it's a longer term project... It would be nice to extend these operations, using for instance https://us.metamath.org/mpeuni/df-bj-mulc.html

Reply:
I did find a shorter proof with `reabsifneg` because it matches the condition in `df-abs`

Reply:
You probably mean "the condition in `df-sgn`" ? Well, I think that the definition of signum is the one that should be changed. That's for later, obviously (but if it's changed while still not much used, it will be easier).